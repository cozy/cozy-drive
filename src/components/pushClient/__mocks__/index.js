export const track = jest.fn()

export const isLinux = jest.fn()
export const isAndroid = jest.fn()
export const isIOS = jest.fn()

export const DESKTOP_BANNER = 'desktop_banner'
export const NOVIEWER_DESKTOP_CTA = 'noviewer_desktop_cta'

export const isClientAlreadyInstalled = jest.fn().mockResolvedValue(false)
