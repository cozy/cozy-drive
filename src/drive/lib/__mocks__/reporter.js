module.exports = {
  configureReporter: jest.fn()
}
