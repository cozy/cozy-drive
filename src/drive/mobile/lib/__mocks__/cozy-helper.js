const mockStackClient = {
  setOAuthOptions: jest.fn(),
  setCredentials: jest.fn()
}

const mock = {
  getLang: jest.fn(),
  initClient: jest.fn(() => ({
    getStackClient: jest.fn(() => mockStackClient)
  })),
  initBar: jest.fn(),
  restoreCozyClientJs: jest.fn(),
  resetClient: jest.fn(),
  getOauthOptions: jest.fn(),
  permissions: []
}

module.exports = mock
