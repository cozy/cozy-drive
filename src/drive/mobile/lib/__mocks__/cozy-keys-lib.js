jest.mock('cozy-keys-lib', () => ({
  withVaultClient: jest.fn().mockReturnValue({}),
  useVaultClient: jest.fn()
}))
