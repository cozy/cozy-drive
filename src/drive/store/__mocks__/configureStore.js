module.exports = () => ({
  getState: jest.fn(() => ({})),
  dispatch: jest.fn()
})
