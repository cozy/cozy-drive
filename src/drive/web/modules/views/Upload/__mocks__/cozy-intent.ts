import { FileFromNative, FileForQueue } from '../UploadTypes'

export const webviewIntent = {
  call: (
    methodName: string,
    files?: FileFromNative[] | FileForQueue[]
  ): Promise<FileFromNative[] | void> => {
    if (methodName === 'getFilesToHandle') {
      return Promise.resolve([
        {
          fileName: 'test.pdf',
          type: 'application/pdf',
          size: 1000,
          lastModified: 123456789
        }
      ])
    }
    if (methodName === 'uploadFiles') {
      return Promise.resolve(files?.map(file => file as FileFromNative))
    }
    if (methodName === 'resetFilesToHandle') {
      return Promise.resolve()
    }
    return Promise.resolve()
  }
}
