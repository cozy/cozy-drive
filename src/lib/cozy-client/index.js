export { default as CozyProvider } from './CozyProvider'
export { default as CozyClient } from './CozyClient'
export { default as cozyConnect } from './connect'
export { default as cozyMiddleware } from './middleware'
export {
  default as reducer,
  makeActionCreator,
  fetchCollection,
  refetchCollections,
  fetchDocument,
  fetchReferencedFiles,
  addReferencedFiles,
  removeReferencedFiles,
  applySelectorForAction,
  enhancePropsForActions,
  getCollection,
  getDocument,
  createDocument,
  updateDocument,
  deleteDocument,
  createFile,
  trashFile,
  CREATE_DOCUMENT
} from './reducer'
export {
  fetchSharings,
  fetchContacts,
  getSharingDetails,
  share,
  unshare,
  leave,
  shareByLink,
  revokeLink
} from './slices/sharings'
export {
  startSync,
  isFirstSync,
  isSynced,
  hasSyncStarted
} from './slices/synchronization'

export { downloadArchive, downloadFile } from './helpers'
