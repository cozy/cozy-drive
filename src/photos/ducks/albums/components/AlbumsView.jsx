import React, { Component } from 'react'
import styles from '../../../styles/layout.styl'
import { translate } from 'cozy-ui/react/I18n'

import AlbumsToolbar from './AlbumsToolbar'
import AlbumsList from './AlbumsList'
import Loading from '../../../components/Loading'
import ErrorComponent from 'components/Error/ErrorComponent'
import Topbar from '../../../components/Topbar'

const Content = ({ list }) => {
  const { fetchStatus } = list
  switch (fetchStatus) {
    case 'pending':
    case 'loading':
      return <Loading loadingType="albums_fetching" />
    case 'failed':
      return <ErrorComponent errorType="albums" />
    default:
      return <AlbumsList {...list} />
  }
}

class AlbumsView extends Component {
  render() {
    const { t } = this.props
    if (this.props.children) return this.props.children
    if (!this.props.albums) {
      return null
    }
    return (
      <div
        data-test-id="album-pho-content-wrapper"
        className={styles['pho-content-wrapper']}
      >
        <Topbar viewName="albums">
          <AlbumsToolbar t={t} />
        </Topbar>
        <Content list={this.props.albums} />
      </div>
    )
  }
}

export default translate()(AlbumsView)
