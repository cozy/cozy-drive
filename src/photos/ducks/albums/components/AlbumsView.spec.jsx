import React from 'react'
import { mount } from 'enzyme'
import PropTypes from 'prop-types'
import AppLike from '../../../../../test/components/AppLike'
import { mockedRouter } from '../../../../../test/__mocks__/mockedRouter'

import AlbumsView from './AlbumsView'
import { createMockClient } from 'cozy-client'

describe('AlbumsView', () => {
  const client = createMockClient({})
  const setup = ({ albums }) => {
    const component = mount(
      <AppLike client={client}>
        <AlbumsView albums={albums} />
      </AppLike>,
      {
        context: {
          router: {
            ...mockedRouter
          }
        },
        childContextTypes: {
          router: PropTypes.object
        }
      }
    )
    return { component }
  }
  it('displays an empty Component', () => {
    const { component } = setup({ albums: { data: [] } })
    expect(component.render()).toMatchSnapshot()
  })
})
