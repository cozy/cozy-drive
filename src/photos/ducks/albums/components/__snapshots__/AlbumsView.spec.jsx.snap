// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumsView displays an empty Component 1`] = `
<div
  class="pho-content-wrapper"
  data-test-id="album-pho-content-wrapper"
>
  <div
    class="pho-content-header"
  >
    <h2
      class="pho-content-title"
      data-test-id="pho-content-title"
    >
      Nav.albums
    </h2>
    <div
      class="pho-toolbar"
      data-test-id="pho-toolbar-albums"
      role="toolbar"
    >
      <div
        class="u-hide--mob"
      >
        <a
          class="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N"
          data-test-id="album-add"
          href="#/albums/new"
        >
          <span>
            <svg
              aria-hidden="true"
              class="styles__icon___23x3R"
              focusable="false"
              height="16"
              width="16"
            >
              <use
                href="#album-add"
              />
            </svg>
            <span>
              Toolbar.album_new
            </span>
          </span>
        </a>
      </div>
      <div
        class="styles__c-menu___2YTiU styles__c-menu--right___3wxbb pho-toolbar-menu u-hide--desk"
      >
        <button
          class="styles__c-btn___-2Vnj styles__c-btn--secondary___3Br_N styles__c-btn--narrow___27FHD styles__c-btn--center___16_Xh dri-btn--more"
          data-test-id="more-button"
          title="More"
          type="submit"
        >
          <span>
            <svg
              aria-hidden="true"
              class="styles__icon___23x3R"
              focusable="false"
              height="16"
              width="16"
            >
              <use
                href="#dots"
              />
            </svg>
            <span
              class="u-visuallyhidden"
            >
              More
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="styles__c-empty___3w5oV"
    data-test-id="empty-folder"
  >
    <svg
      class="styles__c-empty-img___2GC4d styles__icon___23x3R"
      height="100%"
      width="100%"
    >
      <use
        href="#icon-0"
      />
    </svg>
    <h2
      class="u-title-h1 styles__c-empty-title___2HduE"
    >
      empty.albums_title
    </h2>
    <p
      class="styles__c-empty-text___3HnvR"
    >
      empty.albums_text
    </p>
    <div
      class="styles__c-empty-text___3HnvR"
    />
  </div>
</div>
`;
